### DISCUSSION TITLE
baseBranches not working as expected

### DISCUSSION THREAD

**Discussion by ghost** [renovatebot/renovate#8505] [Request Help]
Hello,

If in our default branch `master` we have a Renovate config file that contains a `baseBranches` config other than the default branch (i.e. "baseBranches": ["develop"]). Even if in our 2 branches (`master` and `develop`) we have updates to do, no PRs are created at the end. But if we specify both ["master", "develop"], we have PRs created for the 2 branches.

We noticed that we cannot scan the defaut branch, create a PR with `develop` as target branch. Maybe it's a bug(?)

Do you have the same results on your side? Do we have a workaround to do this?

Thanks

**Comment by rarkins** [comment]
I tested this exact scenario yesterday and it worked as expected. Please reproduce on a minimal repo (eg just one dependency) and save the debug logs.

**Comment by ppussar** [comment]
I have the same usecase with the same outcome. Using the config "baseBranches":["master", "develop"] creates 2 PRs one for master and one for develop.
But I'd like to scan master and create PRs for develop

**Reply by rarkins** [reply]
What you write doesn't seem to make any sense. Please describe in more detail and/or with a reproduction repo so I can understand your use case better

**Reply by ppussar** [reply]
Thanks for your fast reply. 
Maybe I misinterpreted the FAQ section 'Use an alternative branch as my Pull Request target' where it is described that it is possible to use the baseBranches config to set an alternative PR-target branch.

My usecase: I have an expensive CICD Pipeline, which has to use a shared integration test environment. So I dont want to test any single renovate branch through the entire pipeline.
May desired setup would be:
1. Renovate scans the main branch
2. Renovate opens PRs with active automerge and merge-target-branch "upgrade"
3. a minimized Build and Test pipeline checks the PR, but without using the full integration test thing
4. once a week a scheduled pipeline is running on the upgrade branch and merges all upgrades back to the main branch, with full integration test support.

That way I can gather pending upgrades in the upgrade branch and can schedule the usage of the integration test environment.

To build that setup I need the possibility to configure renovate in a way that it scans the main branch, but to use in the pull request another branch as target.

**Reply by viceice** [reply]
That's not possible, but there is an open issue about it.

base branches is always the scan and PR target. The default branch is always the source for the renovate config.

**Reply by rarkins** [reply]
What if we had the option to create the base branch if it doesn't exist? We wouldn't want the complexity of keeping it up to date, but we could branch off default branch once

**Reply by ppussar** [reply]
@viceice thanks for the clarification!

@rarkins I guess that we would still have the update problematic in case of updates on the main-branch. I'll solve it now with an rebase job in my pipeline, which will also create the base-branch in case it does not exists.
